### ISSUE TITLE
document `assert_allclose` and `pytest.approx`

### ISSUE THREAD

**Issue by IAlibay** [MDAnalysis/UserGuide#234]
Since these are the favoured assertion methods for the core library, we should document these in `testing.rst`.

**Comment by jandom** [comment]
This ticket sounds very similar to https://github.com/MDAnalysis/UserGuide/issues/171 – what's the difference? pytest.approch?

**Comment by IAlibay** [comment]
@jandom yeah the addition here is pytest.approx. MDAnalysis recommends using approx for single float comparisons and allclose for arrays (it's purely a speed thing, allclose on single numbers is very slow).
